aiohttp/web_reqrep.py

```diff
@@ -381,6 +381,33 @@ def post(self):
         self._post = MultiDictProxy(out)
         return self._post
 
+    def clone(self, *, method=None, path=None, headers=None, raw_headers=None):
+        """
+        Creates and returns a new instance of Request object. If no parameters
+        are given, an exact copy is returned. If a parameter is not passed, it
+        will reuse the one from the current request object.
+
+        :param method: str http method
+        :param path: str url path to use
+        :param headers: CIMultidictProxy or compatible containing the headers.
+        :param raw_headers: tuple of two element tuples containing the headers
+            as bytearrays.
+        """
+
+        message = RawRequestMessage(
+            method or self.method, path or self.path, self.version,
+            headers or self.headers, raw_headers or self.raw_headers,
+            self.keep_alive, None)
```

Comment:
I was wondering the same... Same with the headers and raw_headers. Why do we need to send both replicated? It would be cleaner to just send the `headers` and do the conversion (if ever needed)